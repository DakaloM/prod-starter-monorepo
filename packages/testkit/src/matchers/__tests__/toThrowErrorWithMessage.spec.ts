class CustomExtendedError extends Error {
  public statusCode = 400;
  constructor(message: string) {
    super(message);
  }
}

describe('test/matchers/toThrowErrorWithMessage', () => {
  test('correctly compares errors', () => {
    expect(() => {
      throw new Error('a');
    }).toThrowErrorWithMessage(Error, 'a');
  });

  test('fails on non matching errors', () => {
    try {
      expect(() => {
        throw new Error('a');
      }).toThrowErrorWithMessage(ExtendedError, 'something else');
    } catch (err) {
      if (err instanceof Error) {
        expect(err.message).toMatchSnapshot(
          `Expected an error of type "BadRequestError" with message:`,
        );
      } else {
        throw err;
      }
    }
  });
});
