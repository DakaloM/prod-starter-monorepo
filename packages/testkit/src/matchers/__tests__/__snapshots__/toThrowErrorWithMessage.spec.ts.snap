// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`test/matchers/toThrowErrorWithMessage > fails on non matching errors > Expected an error of type "BadRequestError" with message: 1`] = `"ExtendedError is not defined"`;
